napari/conftest.py

```diff
@@ -470,3 +472,219 @@ def _mock_app():
             yield app
         finally:
             Application.destroy('test_app')
+
+
+def _get_calling_place(depth=1):
+    if hasattr(sys, "_getframe"):
+        frame = sys._getframe(1 + depth)
+        return f"{frame.f_code.co_filename}:{frame.f_code.co_firstlineno}"
+    return ""
+
+
+@pytest.fixture
+def dangling_qthreads(monkeypatch, qtbot):
+    from qtpy.QtCore import QThread
+
+    base_start = QThread.start
+    thread_dict = WeakKeyDictionary()
+    # dict of threads that have been started but not yet terminated
+
+    def my_start(self, priority=QThread.InheritPriority):
+        thread_dict[self] = _get_calling_place()
+        base_start(self, priority)
+
+    monkeypatch.setattr(QThread, 'start', my_start)
+    yield
+
+    dangling_threads_li = []
+
+    for thread, calling in thread_dict.items():
+        try:
+            if thread.isRunning():
+                dangling_threads_li.append((thread, calling))
+        except RuntimeError as e:
+            if "wrapped C/C++ object of type QThread" not in e.args[0]:
+                raise
+
+    for thread, _ in dangling_threads_li:
+        with qtbot.waitUntil(thread.isFinished, timeout=2000):
```

Comment:
I'm not sure what using `qtbot.waitUntil` as a context manager means. [The docs](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.waitUntil) just show plain calls to it. So maybe you can just call it before/after `quit`?

Reply:
True